Content.Server/Anomaly/AnomalySynchronizerSystem.cs

```diff
@@ -0,0 +1,158 @@
+        SubscribeLocalEvent<AnomalyStabilityChangedEvent>(OnAnomalyStabilityChanged);
+    }
+
+    private void OnPowerChanged(EntityUid uid, AnomalySynchronizerComponent component, ref PowerChangedEvent args)
+    {
+        if (args.Powered)
+            return;
+
+        if (!TryComp<AnomalyComponent>(component.ConnectedAnomaly, out var anomaly))
+            return;
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly);
+        DisconnetFromAnomaly(uid, component, anomaly);
+    }
+
+    private void OnInteractHand(EntityUid uid, AnomalySynchronizerComponent component, InteractHandEvent args)
+    {
+        foreach (var entity in _entityLookup.GetEntitiesInRange(uid, 0.15f)) //is the radius of one tile. It must not be set higher, otherwise the anomaly can be moved from tile to tile
+        {
+            if (TryComp<AnomalyComponent>(entity, out var anomaly))
+            {
+                ConnectToAnomaly(uid, component, anomaly);
+                break;
+            }
+        }
+    }
+
+    private void ConnectToAnomaly(EntityUid uid, AnomalySynchronizerComponent component, AnomalyComponent anomaly)
+    {
+        var auid = anomaly.Owner;
+
+        if (component.ConnectedAnomaly == auid)
+            return;
+
+        component.ConnectedAnomaly = auid;
+        //move the anomaly to the center of the synchronizer, for aesthetics.
+        var targetXform = _transform.GetWorldPosition(uid);
+        _transform.SetWorldPosition(auid, targetXform);
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly);
+        _popup.PopupEntity(Loc.GetString("anomaly-sync-connected"), uid, PopupType.Medium);
+        _audio.PlayPvs(component.ConnectedSound, uid);
+    }
+
+    //TO DO: disconnection from the anomaly should also be triggered if the anomaly is far away from the synchronizer.
+    //Currently only bluespace anomaly can do this, but for some reason it is the only one that cannot be connected to the synchronizer.
+    private void DisconnetFromAnomaly(EntityUid uid, AnomalySynchronizerComponent component, AnomalyComponent anomaly)
+    {
+        if (component.ConnectedAnomaly == null)
+            return;
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly); 
+        _popup.PopupEntity(Loc.GetString("anomaly-sync-disconnected"), uid, PopupType.Large);
+        _audio.PlayPvs(component.ConnectedSound, uid);
+
+        component.ConnectedAnomaly = default!;
+    }
+
+    private void OnAnomalyPulse(ref AnomalyPulseEvent args)
+    {
+        var query = EntityQueryEnumerator<AnomalySynchronizerComponent>();
+        while (query.MoveNext(out var ent, out var component))
+        {
+            if (args.Anomaly != component.ConnectedAnomaly)
+                continue;
+
+            var uid = component.Owner;
+
+            if (!TryComp<ApcPowerReceiverComponent>(uid, out var apcPower))
```

Comment:
change to be trycomp && powered so you can make a prototype that doesnt use power and still works

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/96445749/5e69daad-ff9e-4ecb-97d9-d23784033cee)
something never works for me at all now...

Reply:
whats wrong with that

Reply:
```csharp
private void OnAnomalyPulse(ref AnomalyPulseEvent args)
    {
        var query = EntityQueryEnumerator<AnomalySynchronizerComponent>();
        while (query.MoveNext(out var ent, out var component))
        {
            if (args.Anomaly != component.ConnectedAnomaly)
                continue;
            if (_power.IsPowered(ent)) //<--- THIS
                continue;

            var uid = args.Anomaly;
            _signal.InvokePort(uid, component.PulsePort);
        }
    }
```
this test never succeeds, regardless of the power supply.

Reply:
oh, i fix it